File: Src/CSharpier.VisualStudio/CSharpier.VisualStudioShared/ICSharpierProcess.cs
namespace CSharpier.VisualStudio
{
    public interface ICSharpierProcess
    {
        string Version { get; }
        bool ProcessFailedToStart { get; }
        string FormatFile(string content, string fileName);
        void Dispose();
    }

    public interface ICSharpierProcess2 : ICSharpierProcess
    {
        FormatFileResult? formatFile(FormatFileParameter parameter);
    }

    public class FormatFileParameter
    {
        public string fileContents = string.Empty;
        public string fileName = string.Empty;
    }

    public class FormatFileResult
    {
        public string formattedFile = string.Empty;
        public Status status;
        public string errorMessage = string.Empty;
    }

    public enum Status
    {
        Formatted,
        Ignored,
        Failed,
        UnsupportedFile,
    }
}

File: Src/CSharpier.Rider/src/main/java/com/intellij/csharpier/ICSharpierProcess.java
package com.intellij.csharpier;

interface ICSharpierProcess {
  String getVersion();
  boolean getProcessFailedToStart();
  String formatFile(String content, String fileName);
  void dispose();
}

interface ICSharpierProcess2 extends ICSharpierProcess {
  FormatFileResult formatFile(FormatFileParameter parameter);
  void dispose();
}

class FormatFileParameter {

  public String fileContents;
  public String fileName;
}

class FormatFileResult {

  public String formattedFile;
  public Status status;
  public String errorMessage;
}

enum Status {
  Formatted,
  Ignored,
  Failed,
}
